src/math/ITiglObjectiveFunction.cpp

```diff
@@ -66,13 +67,11 @@ void ITiglObjectiveFunction::getNumericalGradient(const double * x, double * dx)
     for (int i = 0; i < getParameterCount(); ++i) {
         xnew[i] = x[i] + h;
 
-        double ofi = getFunctionValue(xnew);
+        double ofi = getFunctionValue(&xnew.front());
```

Comment:
The benefit in using a vector is that you don't have to care about memory management, the vector allocates the required memory on the heap and automatically releases it as soon as the lifetime of the vector ends (see RAII idiom). As you mentioned this also avoids memory leaks when exceptions are thrown. 
According to C++ Standard ISO/IEC 14882:2003 
> The elements of a vector are stored contiguously, meaning that if v is a vector<T, Allocator> where T is some type other than bool, then it obeys the identity &v[n] == &v[0] + n for all 0 <= n < v.size().

So internally the vector holds an contiguous array of size n on the heap, so the behavior is identical to the original code.